File: backend_api/models/product.js
const mongoose = require('mongoose');

const productSchema = mongoose.Schema({
    productName: {
        type: String,
        required: true,
        trim: true,
    },

    productPrice: {
        type: Number,
        required: true, 
    },

    quantity: {
        type: Number,
        required: true,
    },

    description: {
        type: String,
        required: true,
        trim: true,
    },

    category: {  // ✅ Changed to lowercase
        type: String,
        required: true,
    },

    subCategory: {
        type: String,
        required: true,
    },

    images: [{
        type: String,
        required: true,
    }],

    popular: {
        type: Boolean,
        default: false,
    },

    recommend: {
        type: Boolean,
        default: false,
    }
});

const Product = mongoose.model("Product", productSchema);
module.exports = Product;

File: lib/models/product.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'dart:convert';

class Product {
  final String id;
  final String productName;
  final int productPrice;
  final int quantity;
  final String description;
  final String category;
  final String vendorId;
  final String fullName;
  final String subCategory;
  final List<String> images;

  Product({
    required this.id,
    required this.productName,
    required this.productPrice,
    required this.quantity,
    required this.description,
    required this.category,
    required this.vendorId,
    required this.fullName,
    required this.subCategory,
    required this.images,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'id': id,
      'productName': productName,
      'productPrice': productPrice,
      'quantity': quantity,
      'description': description,
      'category': category,
      'vendorId': vendorId,
      'fullName': fullName,
      'subCategory': subCategory,
      'images': images,
    };
  }

  factory Product.fromMap(Map<String, dynamic> map) {
    return Product(
      id: map['_id'] ?? '',
      productName: map['productName'] ?? '',
       productPrice: (map['productPrice'] as num).toInt(), // ✅ handles both int/double
       quantity: (map['quantity'] as num).toInt(), 
      description: map['description'] ?? '',
      category: map['category'] ?? '',
      vendorId: map['vendorId'] ?? '',
      fullName: map['fullName'] ?? '',
      subCategory: map['subCategory'] ?? '',
      images:
          map['images'] != null
              ? List<String>.from(map['images'].map((e) => e.toString()))
              : [],
    );
  }

  String toJson() => json.encode(toMap());

  factory Product.fromJson(String source) =>
      Product.fromMap(json.decode(source) as Map<String, dynamic>);
}
